error-prone-contrib/src/test/java/tech/picnic/errorprone/bugpatterns/util/ThirdPartyLibraryTest.java

```diff
@@ -86,23 +87,64 @@ void isIntroductionAllowedIgnoreClasspathCompat(boolean ignoreClassPath) {
         .doTest();
   }
 
+  @Test
+  void canIntroduceUsage() {
+    CompilationTestHelper.newInstance(CanIntroduceUsageTestChecker.class, getClass())
+        .addSourceLines(
+            "A.java",
+            "// BUG: Diagnostic contains: GUAVA_PUBLIC: true, GUAVA_PRIVATE: false, ERROR_PRONE_PUBLIC_NESTED:",
+            "// true",
+            "class A {}")
+        .doTest();
+  }
+
   /**
    * Flags classes with a diagnostics message that indicates, for each {@link ThirdPartyLibrary}
    * element, whether they can be used.
    */
   @BugPattern(severity = ERROR, summary = "Interacts with `ThirdPartyLibrary` for testing purposes")
-  public static final class TestChecker extends BugChecker implements ClassTreeMatcher {
+  public static final class IsIntroductionAllowedTestChecker extends BugChecker
+      implements ClassTreeMatcher {
     private static final long serialVersionUID = 1L;
 
     @Override
     public Description matchClass(ClassTree tree, VisitorState state) {
       return buildDescription(tree)
           .setMessage(
               Arrays.stream(ThirdPartyLibrary.values())
+                  .map(lib -> lib.name() + ": " + lib.isIntroductionAllowed(state))
+                  .collect(joining(", ")))
+          .build();
+    }
+  }
+
+  /**
+   * Flags classes with a diagnostics message that indicates, for selected types, the result of
+   * {@link ThirdPartyLibrary#canIntroduceUsage(String, VisitorState)}.
+   */
+  @BugPattern(severity = ERROR, summary = "Interacts with `ThirdPartyLibrary` for testing purposes")
+  public static final class CanIntroduceUsageTestChecker extends BugChecker
+      implements ClassTreeMatcher {
+    private static final long serialVersionUID = 1L;
+    private static final ImmutableMap<String, String> TYPES =
+        ImmutableMap.of(
+            "GUAVA_PUBLIC",
+            ImmutableList.class.getCanonicalName(),
+            "GUAVA_PRIVATE",
```

Comment:
Depends on how you look at it, but strictly speaking this type is not `private`, right?

Reply:
Indeed, it's package-private, but here I meant private-to-the-library/implementation.